lib/ja_serializer/dsl.ex

```diff
@@ -0,0 +1,424 @@
+defmodule JaSerializer.DSL do
+  @moduledoc """
+  A DSL for defining JSON-API.org spec complaint payloads.
+
+  Built on top of the `JaSerializer.Serializer` behaviour.
+
+  The following macros are available:
+
+    * `location/1` - Define the url of a single serialized object.
+    * `attributes/1` - Define the attributes to be returned.
+    * `has_many/2` - Define a has_many relationship.
+    * `has_one/2` - Define a has_one or belongs_to relationship.
+
+  This module should always be used in conjunction with
+  `JaSerializer.Serializer`, see `JaSerializer` for the best way to do so.
+
+  When `use`ing this module default implementations of the `links/2`,
```

Comment:
Is it implied that these are "callbacks"? I got to the next sentence and wasn't clear whether it was talking about links, attributes and relationships or something else. Would it make sense to say that these are callbacks?